### PR TITLE
wizard-rename-hasBodyPadding rule

### PR DESCRIPTION
**Pull request by jschuler**
Closes: https://github.com/patternfly/pf-codemods/issues/62

### CODE REVIEW

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js [hunk 1]
```diff
@@ -1,10 +1,42 @@
...
+    return imports.length == 0 ? {} : {
+      JSXOpeningElement(node) {
+        if (imports.map(imp => imp.local.name).includes(node.name.name)) {
+          const attribute = node.attributes.find(node => node.name && node.name.name === 'hasBodyPadding');
+          if (attribute) {
+            context.report({
+              node,
+              message: 'The Wizard hasBodyPadding prop was removed in favor of hasNoBodyPadding',
+              fix(fixer) {
+                const attrText = sourceCode.getText(attribute);
```
Comment (redallen) [inline packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:19]: I think you can just use `attribute.value` here

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
+                } else {
+                  if (attrValue.indexOf('false') > -1) {
+                    // hasBodyPadding={false} -> replace with hasNoBodyPadding
+                    fixes.push(fixer.replaceText(attribute, 'hasNoBodyPadding'));
+                  } else if (attrValue.indexOf('true') > -1) {
+                    // hasBodyPadding={true} -> remove prop
+                    fixes.push(fixer.remove(attribute));
+                  }
+                }
+                return fixes;
```
Comment (redallen) [inline packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:34]: Can simplify to `return fixer.replaceText(attribute, attrValue.includes('false') ? 'hasNoBodyPadding' : '');`

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js [hunk 3]
```diff
@@ -1,10 +1,42 @@
...
-  create: renameProp(
-    ['Wizard', 'WizardBody', 'WizardToggle'],
-    {'hasBodyPadding': 'hasNoBodyPadding'},
-    node => `hasBodyPadding prop has been removed for ${node.name.name}. Use hasNoBodyPadding instead`
-  )
-};
\ No newline at end of file
+  create: function(context) {
+    const imports = getPackageImports(context, '@patternfly/react-core')
+      .filter(specifier => specifier.imported.name === 'Wizard');
```
Comment (redallen) [inline packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:7]: I know the PR body didn't mention this, but `WizardBody` and `WizardToggle` also had this prop removed. Change to `['Wizard', 'WizardBody', 'WizardToggle'].includes(specifier.imported.name);`

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js [hunk 4]
```diff
@@ -1,10 +1,42 @@
...
+    const sourceCode = context.getSourceCode();
+      
+    return imports.length == 0 ? {} : {
+      JSXOpeningElement(node) {
+        if (imports.map(imp => imp.local.name).includes(node.name.name)) {
+          const attribute = node.attributes.find(node => node.name && node.name.name === 'hasBodyPadding');
+          if (attribute) {
+            context.report({
+              node,
+              message: 'The Wizard hasBodyPadding prop was removed in favor of hasNoBodyPadding',
```
Comment (redallen) [inline packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:17]: Message can be changed back to include `node.name.name`: `hasBodyPadding prop has been removed for ${node.name.name}. Use hasNoBodyPadding instead`

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js (hunk 1 above)
Comment (jschuler) [reply packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:19]: when i change it to that it's not quite working

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js (hunk 2 above)
Comment (jschuler) [reply packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:34]: oh yaa, much simpler thanks :)

File: packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js (hunk 1 above)
Comment (redallen) [reply packages/eslint-plugin-pf-codemods/lib/rules/wizard-rename-hasBodyPadding.js:19]: Ah, I believe the syntax is `sourceCode.getText(attribute.value.expression_`. Can you change it to that so we handle the weird case of `hasBodyPadding= {false}`?

**Pull request merged**